python/tests/as_of_join_tests.py

```diff
@@ -8,291 +8,559 @@
-                     ["S1", "2020-09-01 00:02:10", 361.1],
-                     ["S1", "2020-09-01 00:19:12", 362.1]]
-
-        right_data = [["S1", "2020-08-01 00:00:01", 345.11, 351.12],
-                      ["S1", "2020-08-01 00:01:05", 348.10, 353.13],
-                      ["S1", "2020-09-01 00:02:01", 358.93, 365.12],
-                      ["S1", "2020-09-01 00:15:01", 359.21, 365.31]]
+        leftSchema = StructType(
+            [
+                StructField("symbol", StringType()),
+                StructField("event_ts", StringType()),
+                StructField("trade_pr", FloatType()),
+            ]
+        )
+
+        rightSchema = StructType(
+            [
+                StructField("symbol", StringType()),
+                StructField("event_ts", StringType()),
+                StructField("bid_pr", FloatType()),
+                StructField("ask_pr", FloatType()),
+            ]
+        )
+
+        expectedSchema = StructType(
+            [
+                StructField("symbol", StringType()),
+                StructField("left_event_ts", StringType()),
+                StructField("left_trade_pr", FloatType()),
+                StructField("right_event_ts", StringType()),
+                StructField("right_bid_pr", FloatType()),
+                StructField("right_ask_pr", FloatType()),
+            ]
+        )
+        expectedSchemaNoRightPrefix = StructType(
+            [
+                StructField("symbol", StringType()),
+                StructField("left_event_ts", StringType()),
+                StructField("left_trade_pr", FloatType()),
+                StructField("event_ts", StringType()),
+                StructField("bid_pr", FloatType()),
+                StructField("ask_pr", FloatType()),
+            ]
+        )
+
+        left_data = [
+            ["S1", "2020-08-01 00:00:10", 349.21],
+            ["S1", "2020-08-01 00:01:12", 351.32],
+            ["S1", "2020-09-01 00:02:10", 361.1],
+            ["S1", "2020-09-01 00:19:12", 362.1],
+        ]
+
+        right_data = [
+            ["S1", "2020-08-01 00:00:01", 345.11, 351.12],
+            ["S1", "2020-08-01 00:01:05", 348.10, 353.13],
+            ["S1", "2020-09-01 00:02:01", 358.93, 365.12],
+            ["S1", "2020-09-01 00:15:01", 359.21, 365.31],
+        ]
 
         expected_data = [
-            ["S1", "2020-08-01 00:00:10", 349.21, "2020-08-01 00:00:01", 345.11, 351.12],
-            ["S1", "2020-08-01 00:01:12", 351.32, "2020-08-01 00:01:05", 348.10, 353.13],
+            [
+                "S1",
+                "2020-08-01 00:00:10",
+                349.21,
+                "2020-08-01 00:00:01",
+                345.11,
+                351.12,
+            ],
+            [
+                "S1",
+                "2020-08-01 00:01:12",
+                351.32,
+                "2020-08-01 00:01:05",
+                348.10,
+                353.13,
+            ],
```

Comment:
In cases like this should we skip black formatting?

Reply:
See my response, below.